bn_mp_factors_zero.c

```diff
@@ -0,0 +1,45 @@
+#include "tommath_private.h"
+#ifdef BN_MP_FACTORS_ZERO_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis
+ *
+ * LibTomMath is a library that provides multiple-precision
+ * integer arithmetic as well as number theoretic functionality.
+ *
+ * The library was designed directly after the MPI library by
+ * Michael Fromberger but has been written from scratch with
+ * additional optimizations in place.
+ *
+ * SPDX-License-Identifier: Unlicense
+ */
+
+/* Zero the elements of a factor-array (an array of mp_int's) and realloc to default
+   size in that order */
+#ifdef LTM_USE_EXTRA_FUNCTIONS
+int mp_factors_zero(mp_factors *f)
+{
+   int i, e = MP_OKAY;
+   mp_int *tmp;
+
+   if (f->factors != NULL) {
+      for (i = 0; i < f->length; i++) {
+         mp_zero(&(f->factors[i]));
```

Comment:
that's not required, `mp_clear()` also zeroes the mp before clearing

Reply:
> first part of the review, more to come later or so

Uh oh! ;-)

> that's not required, mp_clear() also zeroes the mp before clearing

Yepp, residue from trying out if `mp_zero` would suffice here (playing with the heap is always costly) but that results in quite a mess which I found out quickly.